lib/dialects/postgres/query.js

```diff
@@ -9,37 +9,64 @@ var Utils = require('../../utils')
   , Promise = require('../../promise')
   , sequelizeErrors = require('../../errors.js');
 
-// Parses hstore fields if the model has any hstore fields.
-// This cannot be done in the 'pg' lib because hstore is a UDT.
-var parseHstoreFields = function(model, row) {
-  Utils._.forEach(row, function(value, key) {
-    if(value === null) return row[key] = null;
-
-    if (model._isHstoreAttribute(key)) {
-      row[key] = hstore.parse(value);
-    }else if (model.attributes[key] && DataTypes.ARRAY.is(model.attributes[key].type, DataTypes.HSTORE)) {
-      var array = JSON.parse('[' + value.slice(1).slice(0,-1) + ']');
-      row[key] = Utils._.map(array, function(v){return hstore.parse(v);});
-    }else{
-      row[key] = value;
-    }
-  });
-};
-
-var parseRangeFields = function (model, row) {
-  Utils._.forEach(row, function (value, key) {
-    if (value === null) return row[key] = null;
-
-    if (model._isRangeAttribute(key)) {
-      row[key] = range.parse(value, model.attributes[key].type);
-    } else if (model.attributes[key] && DataTypes.ARRAY.is(model.attributes[key].type, DataTypes.RANGE)) {
-      var array = JSON.parse('[' + value.slice(1).slice(0, -1) + ']');
-      row[key] = Utils._.map(array, function (v) { return range.parse(v, model.attributes[key].type.type); });
-    } else {
-      row[key] = value;
-    }
+var parseDialectSpecificFields = {
```

Comment:
For objects please move the var statement and the initialization to two seperate lines for easier readability.